### PR TITLE
Helm chart updated for ExternalDNS, rfc2136 provider, in order to support GSS-TSIG authentication configuration.

### PR DESCRIPTION
**Pull request by v-esteves**
Changed helm template for external-dns as well as the values.yaml and values.schema.json to support several authentication types in the provider RFC2136.

The previous version only supported RFC2136 configuration for TSIG, which doesn't work with Windows DNS.

Fixes https://github.com/k8gb-io/k8gb/issues/1061 
Fixes https://github.com/k8gb-io/k8gb/issues/929

### CODE REVIEW

File: chart/k8gb/templates/external-dns/external-dns.yaml [hunk 1]
```diff
@@ -41,4 +41,18 @@ spec:
             cpu: "500m"
         securityContext:
           readOnlyRootFilesystem: true
+        {{- if .Values.rfc2136.rfc2136auth.gssTsig.enabled }}
+        volumeMounts:
+          - mountPath: /etc/krb5.conf
+            name: kerberos-config-volume
+            subPath: krb5.conf
+        {{- end }}
+      {{- if .Values.rfc2136.rfc2136auth.gssTsig.enabled }}
```
Comment (jkremser) [inline chart/k8gb/templates/external-dns/external-dns.yaml:50]: can't this be simplified to one block? The condition is the same, is there an issue with indentation?

**Review by jkremser — COMMENTED** [review]
these changes are also part of your other pr https://github.com/k8gb-io/k8gb/pull/1065 right?

Please name the pull request better. `1061 fix` is very cryptic :D

File: docs/provider_rfc2136.md
Comment (ytsarev) [inline docs/provider_rfc2136.md:1]: Can we link this one in main README in tutorials section?

File: docs/provider_rfc2136.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+| rfc2136.rfc2136auth.tsig.enabled | bool | `false` | Set to True if the DNS server uses TSIG authentication for DNS updates by ExternalDNS |
+| rfc2136.rfc2136auth.tsig.tsigCreds[0].tsig-secret-alg | string | `"hmac-sha256"` | Algorithm used to generate the token for TSIG |
+| rfc2136.rfc2136auth.tsig.tsigCreds[1].tsig-keyname | string | `"externaldns-key"` |  |
+| rfc2136.rfc2136auth.gssTsig.enabled | bool | `false` | Set to True if the DNS server uses GSS-TSIG (Kerberos) authentication for DNS updates by ExternalDNS |
+| rfc2136.rfc2136auth.gssTsig.kerberosConfigMap | string | `"kerberos-configmap"` | When using GSS-TSIG, a ConfigMap with a valid krb5.conf configuration should be provided |
+| rfc2136.rfc2136auth.gssTsig.gssTsigCreds[0].kerberos-username | string | `"ad-user-account"` | AD user account with permissions for DNS updates |
+| rfc2136.rfc2136auth.gssTsig.gssTsigCreds[1].kerberos-password | string | `"ad-user-account-password"` | Passowrd of the AD user account |
+| rfc2136.rfc2136auth.gssTsig.gssTsigCreds[2].kerberos-realm | string | `"REALM.DOMAIN"` | Kerberos REALM that should be used for authentication |
```
Comment (ytsarev) [inline docs/provider_rfc2136.md:18]: Looks like helm values are duplicated with the main README https://github.com/k8gb-io/k8gb/pull/1064/files#diff-a9751a1d587556fed5161a73d263258bfeb7b5c4632430d9ec0d853529539aa9R98. ? Should we avoid duplication to keep it at the same place?

**Review by ytsarev — COMMENTED** [review]
left minor doc comments, thanks a lot for the contribution

File: docs/provider_rfc2136.md (hunk 2 above)
Comment (v-esteves) [reply docs/provider_rfc2136.md]: I've removed those duplicated values and added a reference to the readme with the Helm values description. This way there isn't duplicated text and updates are only in one place.

File: docs/provider_rfc2136.md
Comment (v-esteves) [reply docs/provider_rfc2136.md:1]: Added link as requested

**Review by ytsarev — COMMENTED** [review]
@v-esteves could you please resolve the conflict in readme? Thanks :)

**Comment by ytsarev** [comment]
@v-esteves Is it fully superseded by #1065 ?

**Comment by v-esteves** [comment]
fix #1061 #929

**Comment by v-esteves** [comment]
@ytsarev as discussed today, I've moved all documentation files from #1065 to this one and we'll close that one.
I think we should label this one with "heavy-tests" like it is on #1065

**Comment by ytsarev** [comment]
@v-esteves thanks a lot for the consolidation, it really helps! Can you please resolve the conflict in README.md?

File: docs/deploy_windowsdns.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
+<h1 align="center" style="margin-top: 0;">Using K8GB on Azure</h1>
```
Comment (ytsarev) [inline docs/deploy_windowsdns.md:1]: Can we also link this one in the main `README.md` so it will easy to discover and also appear on the website?

**Review by ytsarev — CHANGES_REQUESTED** [review]
@v-esteves last bit before the merge I believe :)

**Comment by ytsarev** [comment]
Please also check the golic output https://github.com/k8gb-io/k8gb/actions/runs/5819518828/job/15780744306?pr=1064#step:5:84

**Comment by v-esteves** [comment]
@ytsarev I've added the link in main Readme.md has requested.

**Review by ytsarev — CHANGES_REQUESTED** [review]
unfortunately all terratest pipelines are still failing :(  we need to investigate

**Comment by v-esteves** [comment]
@ytsarev can I help with something regarding those falling tests? Let me know what can I do from our end.

**Comment by ytsarev** [comment]
@v-esteves it looks like terratest is failing for the reason, we need to debug the test pipeline. If you can try to validate the local terratest run it will be great.

**Comment by ytsarev** [comment]
image
https://github.com/k8gb-io/k8gb/actions/runs/7209689707?pr=1064
I believe it is actual valid test failure we are facing here.

```
panic: test timed out after 15m0s

goroutine 2511 [running]:
testing.(*M).startAlarm.func1()
/opt/hostedtoolcache/go/1.19.1/x64/src/testing/testing.go:2036 +0x8e
created by time.goFunc
/opt/hostedtoolcache/go/1.19.1/x64/src/time/sleep.go:176 +0x32

```

Something is blocking the test run

File: chart/k8gb/README.md [hunk 4]
```diff
@@ -101,8 +101,15 @@ For Kubernetes `< 1.19` use this chart and k8gb in version `0.8.8` or lower.
 | rfc2136.enabled | bool | `false` |  |
 | rfc2136.rfc2136Opts[0].host | string | `"host.k3d.internal"` |  |
 | rfc2136.rfc2136Opts[1].port | int | `1053` |  |
-| rfc2136.rfc2136Opts[2].tsig-secret-alg | string | `"hmac-sha256"` |  |
-| rfc2136.rfc2136Opts[3].tsig-keyname | string | `"externaldns-key"` |  |
+| rfc2136.rfc2136auth.insecure.enabled | bool | `false` | Set to True if insecure updates to the DNS provided can be executed by ExternalDNS |
```
Comment (ytsarev) [inline chart/k8gb/README.md:104]: @v-esteves there is breaking configuration api change. I would double check if it might break the tests

Comment (v-esteves) [reply chart/k8gb/README.md:104]: Thanks for the tip @ytsarev . Working on it.

**Comment by v-esteves** [comment]
@ytsarev like we have discussed before, the problem was regarding the new values that should be added to the helm statement, in order to deploy correctly External DNS.
To bypass this issue and ensure that there aren't breaking changes, I've added the default value to true for TSIG authentication when provider2136 is active. This will allow that running workloads don't have any issue when upgrading the version. For new installations, the desired configuration should be added through correct values definition.

**Comment by ytsarev** [comment]
@v-esteves great solution with the backward compatibility 👍 Thank you so much!
Please consider creating the issue to track the old values support removal after a couple of project releases

**Review by ytsarev — APPROVED** [review]

**Pull request merged**